main/src/ca/uwaterloo/flix/language/phase/typer/SchemaConstraintGen.scala

```diff
@@ -80,6 +80,28 @@ object SchemaConstraintGen {
     }
   }
 
+  def visitFixpointQueryWithProvenance(e: KindedAst.Expr.FixpointQueryWithProvenance)(implicit c: TypeContext, root: KindedAst.Root, flix: Flix): (Type, Type) = {
+    implicit val scope: Scope = c.getScope
+    e match {
+      case KindedAst.Expr.FixpointQueryWithProvenance(exps, select, withh, tvar, loc1) =>
+        val expsSchemaRows = exps.map(visitExp).map(_._1)
+        val selectSchemaRow = select match {
+          case KindedAst.Predicate.Head.Atom(name, Denotation.Relational, terms, _, loc2) =>
+            val termTypes = terms.map(visitExp).map(_._1)
```

Comment:
Again, something like:

```
val (termTypes, termEffs) = ... unzip
```

but this time assert that all termEffs are pure `(c.expectType(Type.Pure, ...))`